src/Generators/ModelGenerator.php

```diff
@@ -101,18 +101,12 @@ private function fillSoftDeletes($templateData)
         if (!$this->commandData->getOption('softDelete')) {
             $templateData = str_replace('$SOFT_DELETE_IMPORT$', '', $templateData);
             $templateData = str_replace('$SOFT_DELETE$', '', $templateData);
-            $templateData = str_replace('$SOFT_DELETE_DATES$', '', $templateData);
         } else {
             $templateData = str_replace(
                 '$SOFT_DELETE_IMPORT$', "use Illuminate\\Database\\Eloquent\\SoftDeletes;\n",
                 $templateData
             );
-            $templateData = str_replace('$SOFT_DELETE$', infy_tab()."use SoftDeletes;\n", $templateData);
-            $deletedAtTimestamp = config('infyom.laravel_generator.timestamps.deleted_at', 'deleted_at');
-            $templateData = str_replace(
-                '$SOFT_DELETE_DATES$', infy_nl_tab()."protected \$dates = ['".$deletedAtTimestamp."'];\n",
-                $templateData
-            );
+            $templateData = str_replace('$SOFT_DELETE$', "use SoftDeletes;\n", $templateData);
```

Comment:
we are given support to add a custom field for soft delete instant of delete_at only

Reply:
Yes, it`s work.  $SOFT_DELETE_DATES$ is unnecessary. I only removed the type cast.

Reply:
public function initializeSoftDeletes()
    {
        $this->dates[] = $this->getDeletedAtColumn();
    }

https://github.com/laravel/framework/blob/6.x/src/Illuminate/Database/Eloquent/SoftDeletes.php#L34

Reply:
sry, it does not work a custom field without declaring it.

if we use soft delete with a custom key than  must be declared in model